composer/trainer/trainer.py

```diff
@@ -1325,6 +1353,23 @@ def __init__(
             # Only wrap the module if required
             self.state.model = prepare_ddp_module(self.state.model, self._find_unused_parameters)
 
+        # The model would need to be torch.compile()'d after being wrapped in a distributed strategy
+        # to take advantage of any graph breaks.
+        if is_torch_2_0 and not is_model_compiled and compile_config is not None:
+            compiled_model = torch.compile(  # pyright: ignore [reportGeneralTypeIssues]
+                self.state.model, **compile_config)
+            self.state.model = compiled_model._orig_mod
+            self.state.model.forward = compiled_model.dynamo_ctx(self.state.model.forward)
```

Comment:
Should this block, be combined with the logic block @ line 900?

Reply:
According to [this](https://dev-discuss.pytorch.org/t/torchdynamo-update-11-making-fsdp-and-dynamo-work-together/1037) pytorch document, DDP and FSDP wrappers have special optimizations that run with `torch.compile`. Hence, the recommendation is to compile the model after being wrapped in either FSDP or DDP.

Reply:
Would this have stopped us from moving logic block @ line 900 down here?

Alternatively, if a user passes in a compiled model should we throw a value error that tells the user to pass in a regular model + a compile config so they can get the FSDP optimizations?